components/Typography/src/MDCFontScaler.m

```diff
@@ -0,0 +1,337 @@
+        UIContentSizeCategoryAccessibilityExtraLarge : @24,
+        UIContentSizeCategoryAccessibilityExtraExtraLarge : @26,
+        UIContentSizeCategoryAccessibilityExtraExtraExtraLarge : @28
+      };
+      _scalingCurve = scalingCurve;
+    } else if ([MDCTextStyleOverline isEqualToString:textStyle]) {
+      NSDictionary<UIContentSizeCategory, NSNumber *> *scalingCurve = @{
+        UIContentSizeCategoryExtraSmall : @8,
+        UIContentSizeCategorySmall : @8,
+        UIContentSizeCategoryMedium : @9,
+        UIContentSizeCategoryLarge : @10,
+        UIContentSizeCategoryExtraLarge : @12,
+        UIContentSizeCategoryExtraExtraLarge : @14,
+        UIContentSizeCategoryExtraExtraExtraLarge : @16,
+        UIContentSizeCategoryAccessibilityMedium : @18,
+        UIContentSizeCategoryAccessibilityLarge : @20,
+        UIContentSizeCategoryAccessibilityExtraLarge : @22,
+        UIContentSizeCategoryAccessibilityExtraExtraLarge : @24,
+        UIContentSizeCategoryAccessibilityExtraExtraExtraLarge : @26
+      };
+      _scalingCurve = scalingCurve;
+    } else {
+      // If nothing matches, return the metrics for MDCTextStyleBody1
+      NSDictionary<UIContentSizeCategory, NSNumber *> *scalingCurve = @{
+        UIContentSizeCategoryExtraSmall : @13,
+        UIContentSizeCategorySmall : @14,
+        UIContentSizeCategoryMedium : @15,
+        UIContentSizeCategoryLarge : @16,
+        UIContentSizeCategoryExtraLarge : @18,
+        UIContentSizeCategoryExtraExtraLarge : @20,
+        UIContentSizeCategoryExtraExtraExtraLarge : @22,
+        UIContentSizeCategoryAccessibilityMedium : @26,
+        UIContentSizeCategoryAccessibilityLarge : @30,
+        UIContentSizeCategoryAccessibilityExtraLarge : @34,
+        UIContentSizeCategoryAccessibilityExtraExtraLarge : @38,
+        UIContentSizeCategoryAccessibilityExtraExtraExtraLarge : @42
+      };
+      _scalingCurve = scalingCurve;
+    }
+  }
+
+  return self;
+}
+
+- (UIFont *)scalableFontWithFont:(UIFont *)font {
+  // If it is available, query the preferredContentSizeCategory.
+  UIContentSizeCategory sizeCategory = UIContentSizeCategoryLarge;
+  if ([UIApplication mdc_safeSharedApplication]) {
+    sizeCategory = [UIApplication mdc_safeSharedApplication].preferredContentSizeCategory;
+  } else if (@available(iOS 10.0, *)) {
+    sizeCategory = UIScreen.mainScreen.traitCollection.preferredContentSizeCategory;
+  }
+
+  UIFont *scaledFont;
+  if (font.mdc_scalingCurve) {
+    scaledFont = [font mdc_scaledFontForSizeCategory:sizeCategory];
+    scaledFont.mdc_scalingCurve = _scalingCurve;
+  } else {
+    // The original font does not have a scaling curve, so create a copy some we aren't
+    // attaching associated objects to system fonts
+    scaledFont = [font copy];
+    scaledFont.mdc_scalingCurve = _scalingCurve;
+    scaledFont = [scaledFont mdc_scaledFontForSizeCategory:sizeCategory];
+  }
+
+  return scaledFont;
+}
+
+- (CGFloat)scaledValueForValue:(CGFloat)value {
```

Comment:
Please add a comment (preferably doc comment) explaining what this method does. Inputs/outputs are super valuable since the types are opaquely named "value".

Reply:
Added in the header.